File: src/Z.Data/System.Data.DataTable/DataTable.LastRow.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System.Data;

public static partial class Extensions
{
    /// <summary>A DataTable extension method that last row.</summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>A DataRow.</returns>
    public static DataRow LastRow(this DataTable @this)
    {
        return @this.Rows[@this.Rows.Count - 1];
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Data/System.Data.DataTable/DataTable.LastRow.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Data

Public Module Extensions_668

	''' <summary>A DataTable extension method that last row.</summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>A DataRow.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function LastRow(this As DataTable) As DataRow
		Return this.Rows(this.Rows.Count - 1)
	End Function
End Module


